File: app/src/main/java/io/github/tobyhs/weatherweight/api/accuweather/forecasts/Temperature.java
package io.github.tobyhs.weatherweight.api.accuweather.forecasts;

import com.google.auto.value.AutoValue;
import com.google.gson.annotations.SerializedName;
import me.tatarka.gsonvalue.annotations.GsonBuilder;

/**
 * A temperature from AccuWeather's APIs
 */
@AutoValue
public abstract class Temperature {
    /**
     * @return the value of the temperature
     */
    @SerializedName("Value")
    public abstract double getValue();

    @GsonBuilder
    public static Builder builder() {
        return new AutoValue_Temperature.Builder();
    }

    @AutoValue.Builder
    public static abstract class Builder {
        public abstract Builder setValue(double value);
        public abstract Temperature build();
    }
}

File: app/src/main/java/io/github/tobyhs/weatherweight/api/accuweather/forecasts/Temperature.kt
package io.github.tobyhs.weatherweight.api.accuweather.forecasts

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * A temperature from AccuWeather's APIs
 *
 * @property value value of the temperature
 */
@JsonClass(generateAdapter = true)
data class Temperature(
    @Json(name = "Value") val value: Double
)
